datalad/distribution/siblings.py

```diff
@@ -558,7 +579,11 @@ def _configure_remote(
                 lgr.debug("Exception was: %s", exc_str(exc))
 
             if as_common_datasrc:
-                ri = RI(url)
+                # we need a fully configured remote here
+                # do not re-use `url`, but ask for the remote config
+                # that git-annex will use too
+                remote_url = repo.config.get(f'remote.{name}.url')
```

Comment:
to say the truth I am not a "strong user of --as-common-datasrc NAME" and its "directionality" here in relation to `--name`  its own `NAME` value and `--url`... the question is -- can remote `name` be a git annex special remote? if so -- then it most likely would not have a `url` (some like type `git` ones would)

Reply:
No, special remotes do not work. This is creating a `type=git` special remote, which requires a fully set up regular Git remote for `initremote` to work. If there is no URL, it will fail (and should).

The directionality is

```
siblings configure -s GITREMOTE --as-common-datasrc SEPCIALREMOTE
```

`SPECIALREMOTE` will be created based on `GITREMOTE`. You can also use `add` (see tests for both examples), in which case `GITREMOTE` will be created first (requires additional spec of a URL), and then `SPECIALREMOTE` will be created based on it.